### PR TITLE
Twitch API requires client id

### PR DESCRIPTION
**Pull request by DaeronAlagos**
Twitch updated their terms and conditions so all API requests need to be identified by a client id. A user can create their own application at http://dev.twitch.tv which will generate them a client id to use in the module settings.

### CODE REVIEW

File: py3status/modules/twitch_streaming.py [hunk 1]
```diff
@@ -62,7 +65,7 @@ def is_streaming(self):
                 'cached_until': self.py3.CACHE_FOREVER
             }
 
-        r = requests.get('https://api.twitch.tv/kraken/streams/' + self.stream_name)
+        r = requests.get('https://api.twitch.tv/kraken/streams/' + self.stream_name, headers={'Client-ID': self.twitch_client_id})
```
Comment (tobes) [inline py3status/modules/twitch_streaming.py]: @DaeronAlagos It'd be great to get this module working again, if you change this to

```python
        r = requests.get(
            'https://api.twitch.tv/kraken/streams/' + self.stream_name,
            headers={'Client-ID': self.twitch_client_id}
        )
```

it'll fix the flake8 error

File: py3status/modules/twitch_streaming.py [hunk 2]
```diff
@@ -16,6 +16,8 @@
         (default "{stream_name} is offline.")
     stream_name: name of streamer(twitch.tv/<stream_name>)
         (default None)
+    twitch_client_id: Your client id. Create your own key at http://dev.twitch.tv
```
Comment (lasers) [inline py3status/modules/twitch_streaming.py]: (Suggestion) Skip the `twitch_` prefix.

**Comment by DaeronAlagos** [comment]
Thanks for the feedback. I have committed the changes as requested.

File: py3status/modules/twitch_streaming.py [hunk 3]
```diff
@@ -46,13 +56,14 @@ class Py3status:
     format_invalid = "{stream_name} does not exist!"
     format_offline = "{stream_name} is offline."
     stream_name = None
+    client_id = None
```
Comment (lasers) [inline py3status/modules/twitch_streaming.py]: Sort the config parameters and we'll be good to go.

**Comment by DaeronAlagos** [comment]
@tobes (Obviously) I only added the required to make this work, but I'm happy to start from scratch if you prefer it to be to the 2018 standard. Please recommend which module is best to use as an example of the 2018 standard and I'll crack on with getting it redone.

**Comment by tobes** [comment]
@DaeronAlagos I'm happy with it, but as @lasers said you'll need to fix test.

you just need to move `client_id` so it's in alphabetical order and also in the docstring.  Then we should be good to go.  If squashing the commit is too much I'm ok

so like

```python
    cache_timeout = 10
    client_id = None
    format = "{stream_name} is live!"
    format_invalid = "{stream_name} does not exist!"
    format_offline = "{stream_name} is offline."
    stream_name = None
```

@lasers this PR fixes the module so I think we should merge it, and then you can have your fun at leasure

**Comment by DaeronAlagos** [comment]
Damn it. I've borked the squash :( I should know better than to rush this before bed. I'll clean it up tomorrow.

**Comment by tobes** [comment]
@DaeronAlagos I've merged it so you can sleep easy

Thanks for fixing this

**Comment by lasers** [comment]
@DaeronAlagos Thanks. One less issue to worry about. :-)

**Pull request merged**